### PR TITLE
[CHORES] setting up library

### PR DESCRIPTION
**Pull request by Zsaschz**

### CODE REVIEW

File: src/Controllers/InertiaController.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+use Inertia\Inertia;
+use App\Http\Controllers\Controller;
+
+class InertiaController extends Controller
+{
+    public function home()
+    {
+        return Inertia::render('index', [], 'lms-theme::app');
+    }
+}
```
Comment (AldiNFitrah) [inline src/Controllers/InertiaController.php:14]: newline please

File: src/Helpers/Routes.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+        // if the key is not found, return $key
+        if (!isset($controllers[$key])) {
+            return 'Uasoft\\Badaso\\Theme\\LMSTheme\\Controllers\\'.$key;
+        }
+
+        // return the value of the key
+        return $controllers[$key];
+    }
+}
```
Comment (AldiNFitrah) [inline src/Helpers/Routes.php:20]: newline

File: src/Middlewares/HandleInertiaRequests.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     *
+     * @return array
+     */
+    public function share(Request $request)
+    {
+        return array_merge(parent::share($request), [
+
+        ]);
+    }
+}
```
Comment (AldiNFitrah) [inline src/Middlewares/HandleInertiaRequests.php:48]: newline

File: src/Routes/web.php [hunk 4]
```diff
@@ -1 +1,14 @@
...
+use Uasoft\Badaso\Theme\LMSTheme\Controllers\InertiaController;
+
+$lms_route_prefix = config('badaso-lms-theme.lms_theme_prefix');
+
+Route::prefix($lms_route_prefix)
+    ->as('badaso.lms-theme.')
+    ->middleware(['web', HandleInertiaRequests::class])
+    ->group(function () {
+        Route::get('/', [InertiaController::class, 'home']);
+    });
```
Comment (AldiNFitrah) [inline src/Routes/web.php:14]: newline

File: src/resources/app/store/store.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import Vue from "vue";
+import Vuex from "vuex";
+// import api from "./../api";
```
Comment (AldiNFitrah) [inline src/resources/app/store/store.js:3]: perhaps it can be removed if it's unused(?)

File: src/resources/views/app.blade.php [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        use Uasoft\Badaso\Helpers\Config;
+    @endphp
+    {{-- <link rel="shortcut icon" href="{{ Storage::url(Config::get('commerceThemeFavicon')) }}" type="image/x-icon"> --}}
+</head>
+
+<body>
+    @inertia
+</body>
+
+</html>
```
Comment (AldiNFitrah) [inline src/resources/views/app.blade.php:44]: newline

File: src/Tailwind/tailwind.config.js [hunk 7]
```diff
@@ -0,0 +1,11 @@
+module.exports = {
```
Comment (AldiNFitrah) [inline src/Tailwind/tailwind.config.js:1]: is this different with tailwind.config.js in root directory?

File: src/resources/app/store/store.js (hunk 5 above)
Comment (nofamex) [reply src/resources/app/store/store.js:3]: yes it will be cleaner if we remove any unused comment

File: src/Tailwind/tailwind.config.js (hunk 7 above)
Comment (nofamex) [reply src/Tailwind/tailwind.config.js:1]: please check is there any duplicate problem with this and the config in the root folder

**Review by nofamex — APPROVED** [review]
the code is almost the same as the standard previous library so its pretty good for me, i will change the naming convention on the other repo so its matched with this one, also you need to remove redundant file first and excessive comments

File: composer.json [hunk 8]
```diff
@@ -1,15 +1,24 @@
...
-    "name": "nsire/badaso-lms-theme",
-    "autoload": {
-        "psr-4": {
-            "Nsire\\BadasoLmsTheme\\": "src/"
-        }
+  "name": "badaso/lms-theme",
+  "authors": [
+    {
+      "name": "nofamex",
+      "email": "nsiren517@gmail.com"
```
Comment (AldiNFitrah) [inline composer.json:6]: @nofamex changed this one to uasoft's email in this PR https://github.com/uasoft-indonesia/badaso-lms-module/pull/11 . Do we need to change this one as well or just leave it like this?

File: src/Commands/LMSThemeSetup.php [hunk 9]
```diff
@@ -3,7 +3,153 @@
...
+    }
+
+    protected function envListUpload()
+    {
+        return [
+            'POST_THEME_PREFIX' => 'lms',
+        ];
+    }
+
+    protected function addingBadasoEnv()
```
Comment (AldiNFitrah) [inline src/Commands/LMSThemeSetup.php:99]: use verb for function name, e.g. `addBadasoEnv`

File: src/resources/app/pages/index.vue [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+      </v-card-actions>
+    </v-card>
+  </div>
+</template>
+
+<script>
+export default {
+  components: {},
+  data: () => ({
+    posts: "AAcxccccccccccccccccccccccccccA",
```
[the remark this answers is not in the document]
Comment (Zsaschz) [reply src/resources/app/pages/index.vue]: deleted

File: src/Config/badaso-lms-theme.php [hunk 11]
```diff
@@ -1,3 +1,26 @@
...
+    |
+    | This prefix will set the entry point of the badaso commerce theme.
+    |
+    */
+    'theme_prefix' => env('MIX_COMMERCE_WEB_ROUTE_PREFIX', 'lms'),
+
+    /*
+    |--------------------------------------------------------------------------
+    | Override Controller
+    |--------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (Zsaschz) [reply src/Config/badaso-lms-theme.php:17]: yes it is

**Review by AldiNFitrah — APPROVED** [review]

**Pull request merged**